astropy/io/misc/asdf/tags/coords/coords.py

```diff
@@ -0,0 +1,55 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from asdf.yamlutil import custom_tree_to_tagged_tree
+
+from astropy.units import Quantity
+from astropy.coordinates import ICRS, Longitude, Latitude, Angle
+
+from ...types import AstropyType
+from ..unit.quantity import QuantityType
+
+
+__all__ = ['ICRSCoordType']
+
+
+class ICRSCoordType(AstropyType):
+    name = "coords/icrs_coord"
+    types = ['astropy.coordinates.ICRS']
+    requires = ['astropy']
+    version = "1.0.0"
+
+    @classmethod
+    def from_tree(cls, node, ctx):
+        angle = QuantityType.from_tree(node['ra']['wrap_angle'], ctx)
+        wrap_angle = Angle(angle.value, unit=angle.unit)
+        ra = Longitude(
+            node['ra']['value'],
+            unit=node['ra']['unit'],
+            wrap_angle=wrap_angle)
+        dec = Latitude(node['dec']['value'], unit=node['dec']['unit'])
+
+        return ICRS(ra=ra, dec=dec)
```

Comment:
General comment: I don't know whether speed/memory is considered important - will one ever get large arrays of values? -- but copies are made every step here. All classes accept a `copy=False` argument to avoid that.

...  I see below that this is not implemented yet, so no worry about it for now, then.